src/sentry/api/serializers/models/notification_setting.py

```diff
@@ -35,32 +35,41 @@ def get_attrs(
         """
         type_option: Optional[NotificationSettingTypes] = kwargs.get("type")
 
-        # TODO(actorid) This needs to be reworked. `item_list` can either be
-        # a list of Team or User instances. Each item type requires different
-        # query logic for getting notification settings.
-        actor_mapping = {recipient.actor_id: recipient for recipient in item_list}
+        team_map = {}
+        user_map = {}
+        for recipient in item_list:
+            if isinstance(recipient, User):
+                user_map[recipient.id] = recipient
+            if isinstance(recipient, Team):
+                team_map[recipient.id] = recipient
 
         notifications_settings = NotificationSetting.objects._filter(
             type=type_option,
-            target_ids=actor_mapping.keys(),
+            team_ids=list(team_map.keys()),
+            user_ids=list(user_map.keys()),
         )
 
         results: MutableMapping[Union["Team", "User"], MutableMapping[str, Set[Any]]] = defaultdict(
             lambda: defaultdict(set)
         )
 
         for notifications_setting in notifications_settings:
-            target = actor_mapping.get(notifications_setting.target_id)
-            results[target]["settings"].add(notifications_setting)
+            target = None
+            if notifications_setting.user_id:
+                target = user_map[notifications_setting.user_id]
+            if notifications_setting.team_id:
+                target = team_map[notifications_setting.team_id]
+            if target:
+                results[target]["settings"].add(notifications_setting)
 
         for recipient in item_list:
             # This works because both User and Team models implement `get_projects`.
             results[recipient]["projects"] = recipient.get_projects()
 
-            if type(recipient) == Team:
+            if isinstance(recipient, Team):
                 results[recipient]["organizations"] = {recipient.organization}
 
-            if type(recipient) == User:
+            if isinstance(recipient, User):
```

Comment:
Is this also to check against possible subclasses of `Team` and `User`?

Reply:
Yes, this will include subclasses as well.